tests/TestCase/Model/Behavior/UploadBehaviorTest.php

```diff
@@ -442,6 +443,50 @@ public function testAfterDeleteUsesPathProcessorToDetectPathToTheFile()
         $behavior->afterDelete(new Event('fake.event'), $this->entity, new ArrayObject);
     }
 
+    public function testAfterDeleteUsesDeleteCallback()
+    {
+        $this->entity->field = rand(1000, 9999);
+        $path = rand(1000, 9999) . DIRECTORY_SEPARATOR;
+        $methods = array_diff($this->behaviorMethods, ['afterDelete', 'config', 'setConfig', 'getConfig']);
+        $behavior = $this->getMockBuilder('Josegonzalez\Upload\Model\Behavior\UploadBehavior')
+            ->setMethods($methods)
+            ->setConstructorArgs([$this->table, $this->dataOk])
+            ->getMock();
+
+        $this->dataOk['field']['deleteCallback'] = function ($path, $entity, $field, $settings) {
+            return [
+                $path . $entity->{$field},
+                $path . 'sm-' . $entity->{$field},
+                $path . 'lg-' . $entity->{$field}
+            ];
+        };
+
+        $behavior->config($this->dataOk);
+
+        // expecting getPathProcessor to be called with right arguments for dataOk
```

Comment:
Hmm, it's just because I copied code from https://github.com/FriendsOfCake/cakephp-upload/blob/master/tests/TestCase/Model/Behavior/UploadBehaviorTest.php#L415
Will delete the comments.